core/src/main/java/io/dekorate/kubernetes/configurator/ApplyPort.java

```diff
@@ -34,45 +33,71 @@
   private static final String DEFAULT_PATH = "/";
 
   private final Port port;
-  private final List<String> names;
+  private final Map<String, Integer> nameMappings;
 
-  public ApplyPort(Port port, List<String> names) {
-    this.port = port;
-    this.names = names;
+  public ApplyPort(Port port) {
+    this(port, Ports.webPortNameMappings());
   }
 
-  public ApplyPort(Port port, String... names) {
-    this(port, Arrays.asList(names));
+  public ApplyPort(Port port, Map<String, Integer> nameMappings) {
+    this.port = port;
+    this.nameMappings = nameMappings;
   }
 
   @Override
   public void visit(BaseConfigFluent<?> config) {
-    Predicate<PortBuilder> predicate = p -> names.contains(p.getName());
-    if (config.hasMatchingPort(predicate)) {
-      if (Strings.isNotNullOrEmpty(port.getPath()) && !DEFAULT_PATH.equals(port.getPath())) {
-        config.editMatchingPort(predicate)
-            .withPath(port.getPath())
-            .endPort();
-      }
-      if (port.getContainerPort() != 0) {
-        config.editMatchingPort(predicate)
-            .withContainerPort(port.getContainerPort())
-            .endPort();
-      }
-      if ((port.getHostPort() != null) && (port.getHostPort() != 0)) {
-        config.editMatchingPort(predicate)
-            .withHostPort(port.getHostPort())
-            .endPort();
-      } else if (Ports.isWebPort(port)) {
-        config.editMatchingPort(predicate)
-            .withHostPort(80)
-            .endPort();
-      }
-    } else {
-      String name = names.size() > 0 ? names.get(0) : FALLBACK_PORT_NAME;
-      config.addNewPortLike(port)
-          .withName(name)
-          .withContainerPort(port.getContainerPort())
+    Port updated = Ports.populateHostPort(port);
+    Predicate<PortBuilder> matchingPortName = p -> updated.getName().equals(p.getName());
+    Predicate<PortBuilder> matchingHostPort = p -> updated.getHostPort() != null
+        && updated.getHostPort().equals(p.getHostPort());
+
+    boolean matchFound = false;
+    if (config.hasMatchingPort(matchingPortName)) {
+      matchFound = true;
+      applyPath(config, updated, matchingPortName);
+      applyContainerPort(config, updated, matchingPortName);
+
+    }
+
+    if (config.hasMatchingPort(matchingHostPort)) {
+      matchFound = true;
+      applyPath(config, updated, matchingPortName);
+      applyContainerPort(config, updated, matchingPortName);
```

Comment:
shouldn't be `applyContainerPort(config, updated, matchingHostPort);` instead?

Reply:
Great catch!